USSF-ORBIT/ussf-portal-client
# feat: Guardian Ideal event tracking

# SC-1547



## Proposed changes
Add the following event trackers to the portal:
- Add Ideal
- Remove Ideal
- Click next/previous in Guardian Ideal carousel


---



## Reviewer notes
The PR for adding the Guardian Ideal carousel should be merged first, so give that a review if you haven't: https://github.com/USSF-ORBIT/ussf-portal-client/pull/938

To test:
- Run the app locally and bring up your browser's console
- Add a Guardian Ideal widget to the user's My Space and verify that you see the output `ANALYTICS: trackEvent Guardian Ideal Carousel Click on add Ideal carousel Add Ideal` in your browsers console
- Click next/previous in the carousel and verify that you see `ANALYTICS: trackEvent Guardian Ideal Carousel View next/previous Click next/previous`
- Same thing for removing the Guardian Ideal. The console output when removing should be `ANALYTICS: trackEvent Guardian Ideal Carousel Click on remove Ideal carousel Remove Ideal`

Note: I don't know for sure if we have an instance of Matomo running in dev, but if it's possible to test in another environment that would be beneficial.


## Setup
Run the client as usual. The CMS can run on `main`:
- In the CMS repo `yarn services:up && yarn dev`
- In the portal `yarn dev`


---

## Code review steps

### As the original developer, I have

  - ... 
  - [ ] Including cypress-axe checks when UI changes
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
- For any new migrations/schema changes:

### As code reviewer(s), I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a designer reviewer, I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

## Screenshots
